Added the ability to return data from a raised intent (#495)
finos/FDC3: docs/api/ref/Metadata.md

```ts
interface IntentResolution {
/**
   * The application that resolved the intent.
   */
  readonly source: TargetApp;
  /**
   * The version number of the Intents schema being used.
   */
  readonly version?: string;

>   /**
>    * Retrieves a promise that will resolve to data returned by the
>    * application that resolves the raised intent. The promise will 
>    * reject if an error is thrown by the intent handler or the promise
>    * returned by the intent handler is reject. If the intent handler 
>    * does not return a promise this function will return null.
>    */
>   getResult(): Promise<Context> | null;

Author: ```suggestion
  /**
   * Retrieves a promise that will resolve to data returned by the
   * application that resolves the raised intent. If an error is thrown 
   * by the handler function,  the promise returned is rejected, or 
   * no promise is returned then the Desktop Agent MUST reject 
   * the promise returned by the `getResult()` function of the 
   * `IntentResolution` with a string from the `DataError` enumeration.
   */
  getResult(): Promise<Context> | null;
```

```ts
interface IntentResolution {
/**
   * The application that resolved the intent.
   */
  readonly source: TargetApp;
  /**
   * The version number of the Intents schema being used.
   */
  readonly version?: string;
  /**
   * Retrieves a promise that will resolve to data returned by the
   * application that resolves the raised intent. If an error is thrown 
   * by the handler function,  the promise returned is rejected, or 
   * no promise is returned then the Desktop Agent MUST reject 
   * the promise returned by the `getResult()` function of the 
   * `IntentResolution` with a string from the `DataError` enumeration.
   */

>   getResult(): Promise<Context> | null;

Author: ```suggestion
  getResult(): Promise<Context>;
```

   * chose in response to `fdc3.raiseIntentForContext()`.
   */
  readonly intent: string;
  /**
   * The version number of the Intents schema being used.
   */
  readonly version?: string;

>   /**
>    * Retrieves a promise that will resolve to data returned by the
>    * application that resolves the raised intent. If an error is thrown 
>    * by the handler function,  the promise returned is rejected, or 
>    * no promise is returned then the Desktop Agent MUST reject 
>    * the promise returned by the `getResult()` function of the 
>    * `IntentResolution` with a string from the `DataError` enumeration.
>    */
>   getResult(): Promise<Context>;

Author: ```suggestion
  /**
   * Retrieves a promise that will resolve to data returned by the
   * application that resolves the raised intent. If an error occurs 
   * (i.e. an error is thrown by the handler function,  the promise 
   * returned by the handler function is rejected, or no promise is 
   * returned) then the Desktop Agent MUST reject the promise 
   * returned by the `getResult()` function of the `IntentResolution` 
   * with a string from the `DataError` enumeration.
   */
  getResult(): Promise<Context>;
```
